### PR TITLE
Add size option, update readme

### PR DESCRIPTION
**Pull request by artemkosenko**
Hello there. 

I propose two things:
1) An additional boolean configuration parameter 'adjustOverlaps'.
The reason for that is that I have more 'out-connections' than 'in-connections'. (basically elements may fall into multiple categories)
image
It doesn't look nice.
So the 'adjustOverlaps === true' param does the following:
image

By default it is set to 'false'. The usual logic stays the same.

2) An updated way to calculate the largest node.
This is a controversian one. But it helped me.
The problem was due to the described above thing - more 'out-connections' than 'in-connections'.
The 'findLargestNode' returned a node from a middle which then led to a visual bug.
It got:
image
But i expected one of the images above.

I added the 'tree.html' file which can reproduce the issue if the 'findLargestNode' function is reverted back.
Also there is a test for that. 
`expect(oneThenTwo).toEqual(jasmine.objectContaining({y: 28}));`
It failed here because 'y' was undefined;

The controvercial thing is that a few tests are broken because charts do not conform to the png images provided.
I'd be nice you if could help with those images.

Thanks

### CODE REVIEW

**Comment by artemkosenko** [comment]
I realized that I needed to use node priorities to get what i wanted.
So I reverted back the second change.

**Review by kurkle — COMMENTED** [review]
I don't like the `adjustOverlap` as a option name.

What do you think about `size: 'min'` (default 'max')? Should work both ways and can be determined like `Math[options.size](node.in, node.out)`

That would determine the node height. The flows would then be aligned using a simple multiplier (in/height or out/height), which can always be used, and should be limited to be <=1.

File: test/specs/controller.spec.js [hunk 1]
```diff
@@ -70,4 +70,110 @@ describe('controller', function() {
     expect(so).toEqual(jasmine.objectContaining({x: 2}));
     expect(ag).toEqual(jasmine.objectContaining({x: 3}));
   });
+
+  it('should parse a complex flow', function() {
```
Comment (kurkle) [inline test/specs/controller.spec.js:74]: Also could you split this and the fix (not sure what it is currently) for it in separate PR?

**Review by kurkle — APPROVED** [review]
I'll need to do some testing before releasing this, but I'll be away for couple of weeks. So be patient.

**Comment by artemkosenko** [comment]
Sure. Take you time.

**Pull request merged**